controllers/clustergroupupgrade_controller.go

```diff
@@ -149,11 +149,19 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 		return
 	}
 
-	// Verify that the selected clusters are all valid
-	// If this has already been done we will skip it on further reconcile loops
-	if !meta.IsStatusConditionTrue(clusterGroupUpgrade.Status.Conditions, string(utils.ConditionTypes.ClustersSelected)) {
+	// This may be empty for the first reconcile but that's fine since it will get overwritten by the start of the cgu
+	clusters := r.getClustersListFromRemediationPlan(clusterGroupUpgrade)
+	progressingCondition := meta.FindStatusCondition(clusterGroupUpgrade.Status.Conditions, string(utils.ConditionTypes.Progressing))
+
+	// Set the progressing condition if it is not present
+	if progressingCondition == nil || (progressingCondition.Status == metav1.ConditionFalse && progressingCondition.Reason != string(utils.ConditionReasons.Completed)) {
+
 		var reconcile bool
-		reconcile, err = r.validateCR(ctx, clusterGroupUpgrade)
+		if len(clusters) == 0 {
+			clusters, reconcile, err = r.validateCR(ctx, clusterGroupUpgrade)
+		} else {
+			_, reconcile, err = r.validateCR(ctx, clusterGroupUpgrade)
```

Comment:
This means we don't overwrite it as soon as the plan has been populated?

Reply:
If the clusters list is defined here then it was derived from the remediation plan already. And after checking further, getSuccessfulClusters will not return clusters where backup or precaching is in progress which was what was causing my issues with the empty lists before